src/environment-loader/index.ts

```diff
@@ -1,22 +1,34 @@
 import ContainerManager from '../container-manager';
-import {getTokenAddress, validateTokens} from '../token/util';
-import Web3 from 'web3';
-import {LuminoNode, NodeList} from '../types/node';
+import {validateTokens} from '../token/util';
+import {NodeList} from '../types/node';
 import {SetupJson, SetupNode, SetupToken} from '../types/setup';
 import {LuminoTestEnvironment} from "../types/lumino-test-environment";
 
 export default class EnvironmentLoader {
 
+    private containerManager: ContainerManager = ContainerManager.create();
     private nodes: NodeList = {};
 
-    private constructor(private containerManager: ContainerManager) {}
+    private constructor() {}
 
     public static async load(setup: SetupJson): Promise<LuminoTestEnvironment> {
-        const containerManager = await ContainerManager.create('regtest');
-        const setupLoader = new EnvironmentLoader(containerManager);
-        await setupLoader.loadLuminoNodes(setup);
-        await setupLoader.openChannels(setup);
-        return new LuminoTestEnvironment(setupLoader);
+        const environmentLoader = new EnvironmentLoader();
+        await environmentLoader.parseAndLoad(setup);
+        return new LuminoTestEnvironment(environmentLoader);
+    }
+
+    private async parseAndLoad(setup: SetupJson): Promise<void> {
```

Comment:
I don't think we are actually parsing here. The json SetupJson is an already parsed object in memory

Reply:
Yes but you are taking those values and creating instances, so in some way you are parsing the json config to a fully loaded environment.